src/main/java/screens/feature_6_frameworks_drivers/SessionGateway.java

```diff
@@ -0,0 +1,119 @@
+    private final String file;
+    public SessionGateway(String file) {
+        this.file = file;
+    }
+    /**
+     * Returns a string representation of the JSON file it reads.
+     * @return String
+     */
+    public String fileToString() throws IOException {
+        // Files.readString(Path.of("src/main/java/screens/courses_cleaned.json"));
+        return Files.readString(Path.of(this.file));
+    }
+    /**
+     * Given a string representation of a JSON file, return a HashMap of all course info from the JSON file
+     * where the key is the course code and value is course information.
+     * @param jsonData JSON data
+     * @return HashMap<String, CalendarCourse>
+     */
+    public HashMap<String, CalendarCourse> readFromFile(String jsonData) throws ParseException {
+        HashMap<String, CalendarCourse> allCourses = new HashMap<>();
+        // Parse String
+        JSONParser parser = new JSONParser();
+        JSONObject jsonObj = (JSONObject) parser.parse(jsonData);
+        // Iterate through jsonObj
+        for (Object key : jsonObj.keySet()) {
+            JSONObject courseInfo = (JSONObject) jsonObj.get(key);
+            // Iterate through sections
+            JSONObject sections = (JSONObject) courseInfo.get("sections"); // Looks at value of "sections"
+            ArrayList<Section> allSections = new ArrayList<>();
+            for (Object section : sections.keySet()) {
+                JSONObject aSection = (JSONObject) sections.get(section);
+                // Iterate through blocks
+                ArrayList<Block> allBlocks = new ArrayList<>();
+                JSONObject blocks = (JSONObject) aSection.get("blocks"); // Looks at value of "blocks"
+                for (Object block : blocks.keySet()) {
+                    JSONObject aBlock = (JSONObject) blocks.get(block);
+                    Block resultBlock = readFromFileHelper(aBlock);
+                    if (resultBlock != null) {
+                        allBlocks.add(resultBlock);
+                    }
+                }
+                // Pass to SectionBuilderInteractor
+                allSections.add(new SectionBuilderInteractor((String) aSection.get("section"),
+                        (String) aSection.get("prof"), allBlocks).newSection());
+            }
+            // Pass to CourseBuilderInteractor and add to all courses
+            CalendarCourse calCourse = new CalendarCourseBuilderInteractor((String) courseInfo.get("title"),
+                    allSections, (String) courseInfo.get("session"),
+                    (String) courseInfo.get("code"), (String) courseInfo.get("breadth")).newCourse();
+            allCourses.put((String) courseInfo.get("code"), calCourse);
+        }
+        // Return all courses in the JSON file
+        return allCourses;
+    }
+    /**
+     * Helper method of readFromFile to check for conditions of block's values and builds block
+     */
+    public Block readFromFileHelper(JSONObject aBlock) {
```

Comment:
There is a more efficient way to do this without brute forcing through every possibility